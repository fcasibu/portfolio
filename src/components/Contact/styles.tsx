import styled, { keyframes } from 'styled-components';

const slideIn = keyframes`
  from {
    transform: translateX(-500px);
  }

  to {
    transform; translateX(0);
  }
`;

const fadeIn = keyframes`
  from {
    transform: translateY(40px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const StyledContact = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1.2rem;
  background: var(--theme-background-secondary);
  padding: 1rem 1.5rem;
  width: 600px;
  animation: ${slideIn} 0.5s ease-in-out forwards;

  & > * {
    opacity: 0;
    animation: ${fadeIn} 1.2s ease-in-out forwards 0.8s;
  }
@media (max-width: 700px) {
    width: 100%;
    height: 450px;
  }

`;

export const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  gap: 1rem;
  width: 100%;

  > button {
    align-self: flex-end;
  }
`;

export const Wrapper = styled.div`
  display: flex;
  gap: 1rem;
`;

export const Input = styled.input`
  background: var(--theme-background-tertiary);
  color: var(--theme-text-color);
  border: 0;
  outline: 0;
  width: 100%;
  padding: 0.3rem;
`;

export const TextArea = styled.textarea`
  background: var(--theme-background-tertiary);
  color: var(--theme-text-color);
  border: 0;
  outline: 0;
  width: 100%;
  height: 150px;
  padding: 0.3rem;
  resize: none;
`;
