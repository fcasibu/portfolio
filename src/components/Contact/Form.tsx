import * as React from 'react';
import axios from 'axios';
import Button from '../Button';
import { Input, StyledForm, TextArea, Wrapper } from './styles';

export const Form = () => {
  const [inputField, setInputField] = React.useState({
    name: '',
    email: '',
    message: ''
  });
  const [loading, setLoading] = React.useState(false);

  const handleSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    setLoading(true);
    axios
      .post('https://getform.io/f/ee4eac93-eaf3-48ec-89b3-f8f7a991d526', {
        name: inputField.name,
        email: inputField.email,
        message: inputField.message
      })
      .then(() => {
        setInputField({ name: '', email: '', message: '' });
        setLoading(false);
        alert('Your message has been sent!');
      });
  };

  const handleChange = (event: React.FormEvent) => {
    const { name, value } = event.target as HTMLInputElement;
    setInputField((state) => ({
      ...state,
      [name]: value
    }));
  };

  return (
    <StyledForm onSubmit={handleSubmit}>
      <Wrapper>
        <Input
          name="name"
          type="text"
          placeholder="Name"
          value={inputField.name}
          onChange={handleChange}
          required
        />
        <Input
          name="email"
          type="email"
          placeholder="Email"
          value={inputField.email}
          onChange={handleChange}
          required
        />
      </Wrapper>
      <TextArea
        name="message"
        placeholder="Message"
        value={inputField.message}
        onChange={handleChange}
        required
      />
      <Button size="medium" type="submit">
        {loading ? 'Sending...' : 'Submit'}
      </Button>
    </StyledForm>
  );
};
