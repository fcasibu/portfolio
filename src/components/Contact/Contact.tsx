import { Para } from '../Para';
import { Title } from '../Title';
import { Form } from './Form';
import { StyledContact } from './styles';

const Contact = () => {
  return (
    <StyledContact>
      <Title>Contact Me</Title>
      <Para>
        I’m very interested in working with projects. If you have any questions or want to work
        together, don’t hesitate to contact me using the form below. I will get back to you as soon
        as possible.
      </Para>
      <Form />
    </StyledContact>
  );
};

export default Contact;
