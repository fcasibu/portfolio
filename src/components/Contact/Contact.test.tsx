import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import axios from 'axios';
import Contact from './Contact';
import { Form } from './Form';
import userEvent from '@testing-library/user-event';

window.alert = jest.fn();

jest.mock('axios');

const url = 'https://getform.io/f/ee4eac93-eaf3-48ec-89b3-f8f7a991d526';

describe('Contact', () => {
  it('should render and match the snapshot', () => {
    const { container } = render(<Contact />);

    expect(container).toMatchSnapshot();
  });

  it('should be able to send the form on submit', () => {
    const mockedAxios = axios as jest.Mocked<typeof axios>;

    const data = { name: '', email: '', message: '' };
    render(<Form />);

    mockedAxios.post.mockImplementationOnce(() => Promise.resolve(data));
    const input = screen.getAllByRole('textbox');
    const btn = screen.getByRole('button');

    userEvent.type(input[0], 'test');
    userEvent.type(input[1], 'test@test.com');
    userEvent.type(input[2], 'Hello World!');

    userEvent.click(btn);

    expect(screen.getByText('Sending...')).toBeInTheDocument();
    expect(mockedAxios.post).toHaveBeenCalledWith(url, {
      name: 'test',
      email: 'test@test.com',
      message: 'Hello World!'
    });
  });
});
