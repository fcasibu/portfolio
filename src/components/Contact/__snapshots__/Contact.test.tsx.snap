// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render and match the snapshot 1`] = `
<div>
  <div
    class="sc-hKMtZM jHvUUA"
  >
    <h2
      class="sc-gsnTZi fWNryp"
    >
      Contact Me
    </h2>
    <p
      class="sc-bczRLJ iGcDty"
    >
      I’m very interested in working with projects. If you have any questions or want to work together, don’t hesitate to contact me using the form below. I will get back to you as soon as possible.
    </p>
    <form
      class="sc-eCYdqJ ccCzcK"
    >
      <div
        class="sc-jSMfEi fgdzB"
      >
        <input
          class="sc-gKXOVf eMtpPN"
          name="name"
          placeholder="Name"
          required=""
          type="text"
          value=""
        />
        <input
          class="sc-gKXOVf eMtpPN"
          name="email"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
      </div>
      <textarea
        class="sc-iBkjds ePFwPY"
        name="message"
        placeholder="Message"
        required=""
      />
      <button
        class="sc-dkzDqf epFtDs"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
